File: Leetcode/Trees/MaximumDepth.cs
public class Solution
{
    // Iterative solution
    public int MaxDepth(TreeNode root)
    {
        if (root == null)
        {
            return 0;
        }

        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        int depth = 0;

        while (queue.Any())
        {
            int len = queue.Count;
            for (int i = 0; i < len; i++)
            {
                TreeNode item = queue.Dequeue();
                if (item.left != null) queue.Enqueue(item.left);
                if (item.right != null) queue.Enqueue(item.right);
            }
            depth++;
        }
        return depth;
    }

    // Recursive solution
    public int MaxDepth(TreeNode root)
    {
        if (root == null)
        {
            return 0;
        }
        return 1 + Math.Max(MaxDepth(root.left), MaxDepth(root.right));
    }
}

File: Leetcode/Trees/Maximum Depth.js
//104. Maximum Depth of Binary Tree

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    if (!root) {
        return 0;
    }
    let left = maxDepth(root.left)
    let right = maxDepth(root.right)
    return Math.max(left,right) + 1
};

var maxDepth = function(root) {
    if (!root) {
        return 0;
    }
    const queue = [root];
    let depth = 0;
    while (queue.length !== 0) {
        const len = queue.length;
        for (let i = 0; i < len; i++) {
            let item = queue.shift()
            if (item.left) queue.push(item.left);
            if (item.right) queue.push(item.right);
        }
        depth++;        
    }
    return depth;
};